tm2/pkg/store/types/options.go

```diff
@@ -38,3 +38,24 @@ var (
 	// PruneSyncable means only those states not needed for state syncing will be deleted (keeps last 100 + every 10000th)
 	PruneSyncable = NewPruningOptions(100, 10000)
 )
+
+type PruneStrategy string
+
+const (
+	PruneEverythingStrategy PruneStrategy = "everything"
+	PruneNothingStrategy    PruneStrategy = "nothing"
+	PruneSyncableStrategy   PruneStrategy = "syncable"
+)
+
+// Options returns the corresponding prune options.
+// If the pruning strategy is invalid, defaults to no pruning.
+func (s PruneStrategy) Options() PruningOptions {
```

Comment:
Have you seen `NewPruningOptionsFromString()` here  ?https://github.com/gnolang/gno/blob/f6b725541d595bafeffafe58bb59cbdac9cb3c83/tm2/pkg/store/store.go#L14-L26 ?

Maybe you should keep only one of the two.

Reply:
Good catch, I didn't even see this one. It's unused so I dropped it in preference of the new one, and changed the default option to `Syncable`:

[617e23f](https://github.com/gnolang/gno/pull/4340/commits/617e23ff72aba77992ea7a39664517df1099f4ec)